Design/ThirdpartyDependencies.md

```diff
@@ -0,0 +1,300 @@
+is settled.
+
+On Linux this would also have the advantage of supporting additional Linux
+distributions or newer versions of existing distributions almost trivially.
+
+#### CMake
+
+Conan has built in support for
+[CMake](https://docs.conan.io/en/latest/integrations/cmake.html)
+along with a [CMake Wrapper](https://github.com/conan-io/cmake-conan/)
+to enable configuring dependencies completely in CMake.
+
+Developers would continue to work as they do now with their favourite IDEs
+and tools but would need to install Conan first.
+Release packages are [provided](https://github.com/conan-io/conan)
+that have no other dependencies and are easy to install.
+
+#### Python Packages
+
+Conan was designed as a framework to package C/C++ dependencies.
+While it is written in Python itself it does not possess the capability
+to manage Python dependencies. Windows/macOS already ship a bundled Python
+distribution so maintaining a separate site-packages is already supported here.
+It is proposed that the same be done for the desktop application bundle
+on Linux so that it is fully self-contained.
+
+The Windows/macOS Python-bundling solution is clunky.
+Another option is creating our own Python Conan package
+that can be used like any other dependency for the desktop
+application bundle.
+This has been [prototyped](https://github.com/martyngigg/conan-python3)
+and seems to work well.
+It would require some patching of `site.py` or using `sitecustomize.py`
+to ensure packaging doesn't interfere with system versions
+but this is something supported by Python
+and done by other systems like `Conda`.
+Python requirements could be easily supported by a requirements file
+or some newer system like
+[pipenv](https://packaging.python.org/key_projects/#pipenv) to properly
+define a frozen set of dependencies.
+Some packages such as `PyQt` do not provide the headers and link
+libraries required for the C++ build.
+The easiest solution would be a Conan package for `PyQt` much like macOS
+Homebrew does. 
+
+The availability of `mantid` to other Python libraries would be the cause
+of the most concern if we adopted the Conan approach as it is
+about creating a self-contained package.
+
+
+### Conda
+
+*"A conda package is a binary tarball containing
+system-level libraries,
+Python modules,
+executable programs,
+or other components"*.
+
+[Conda](https://docs.conda.io/en/latest/)
+is another cross-platform package manager but with support for a
+variety of languages including Python and C++.
+Alongside `pip` many popular Python packages support distribution through Conda.
+In fact `mantid` already has [Conda packages](https://anaconda.org/mantid/repo)
+for workbench and the framework for Linux and Mac.
+
+[Anaconda](https://repo.continuum.io/pkgs/) hosts many prebuilt packages for
+various combinations of compilers and operating systems.
+[Conda build](https://conda.io/projects/conda-build/en/latest/resources/commands/conda-build.html)
```

Comment:
Compilers are somewhat limited. Getting latest versions of compilers on all platforms is problematic/hacky at best.